integration/test_timescaledb.py

```diff
@@ -294,15 +286,375 @@ async def test_timescaledb_async_emit(timescaledb):
+
+    cursor.execute(
+        f"""
+        CREATE TABLE {schema_table} (
+            time TIMESTAMPTZ NOT NULL,
+            required_col INTEGER NOT NULL,
+            optional_col VARCHAR(50)
+        );
+    """
+    )
+    cursor.execute(f"SELECT create_hypertable('{schema_table}', 'time');")
     cursor.close()
 
-    assert count == 2, "Async emission should insert exactly 2 rows"
+    time_format = "%d/%m/%y %H:%M:%S UTC%z"
+    controller = build_flow(
+        [
+            SyncEmitSource(),
+            TimescaleDBTarget(
+                dsn=dsn_url,
+                table=schema_table,  # Using schema.table format
+                time_col="time",
+                columns=["required_col", "optional_col"],
+                time_format=time_format,
+                max_events=10,
+            ),
+        ]
+    ).run()
+
+    # Test with missing required column
+    invalid_data = {
+        "time": "18/09/19 01:55:10 UTC+0000",
+        "optional_col": "test_value",
+        # Missing required_col
+    }
+
+    # Should raise ValueError for missing required column, showing correct schema.table format
+    with pytest.raises(ValueError, match=r"Missing required non-nullable column 'required_col'"):
+        controller.emit(invalid_data)
+        controller.terminate()
+        controller.await_termination()
+
+    # Cleanup
+    with connection.cursor() as cursor:
+        cursor.execute(f"DROP TABLE IF EXISTS {schema_table} CASCADE;")
+        cursor.execute(f"DROP SCHEMA IF EXISTS {schema_name} CASCADE;")
+
+
+@pytest.mark.asyncio
+async def test_timescaledb_schema_caching(timescaledb):
+    """Test that schema information is properly cached to avoid repeated queries"""
+    (_, _, _, _, _, _, table_name, dsn_url, timestamp_precision, columns_config) = timescaledb
+
+    time_formats = {"milliseconds": "%d/%m/%y %H:%M:%S.%f UTC%z", "microseconds": "%d/%m/%y %H:%M:%S.%f UTC%z"}
+    time_format = time_formats.get(timestamp_precision, "%d/%m/%y %H:%M:%S UTC%z")
+
+    target = TimescaleDBTarget(
+        dsn=dsn_url,
+        table=table_name,
+        time_col="time",
+        columns=columns_config,
+        time_format=time_format,
+        max_events=10,
+    )
+
+    # Initialize the target
+    await target._async_init()
+
+    # First call to get schema should query the database
+    schema1 = await target._get_table_schema()
+    assert isinstance(schema1, dict)
+    assert len(schema1) > 0
+
+    # Second call should return cached result (same object)
+    schema2 = await target._get_table_schema()
+    assert schema1 is schema2  # Should be the same object (cached)
+
+    # Verify schema contains expected information
+    assert "time" in schema1
+    assert "binary_col" in schema1
+    assert schema1["time"]["nullable"] is False  # time column is NOT NULL
+    assert schema1["binary_col"]["nullable"] is True  # binary_col allows NULL
+
+    # Cleanup
+    await target._terminate()
```

Comment:
The cleanup block will not run in case of a test failure.

Reply:
will fix